app/models/types/patterns/collection.rb

```diff
@@ -50,8 +55,12 @@ def all_enabled
         patterns.select { |_, pattern| pattern.enabled? }
       end
 
-      def [](value)
-        patterns.fetch(value)
+      def [](name)
+        patterns[name]
+      end
+
+      def updated(name, pattern)
```

Comment:
WDYT of the name `update_pattern`? We need the `:enabled` value too.

Reply:
The enabled value is part of the pattern. I.e. `pattern` is expected to be a hash in the form `{ enabled:, blueprint: }`.

`update_pattern` sounds good to me. I was honestly unsure about this method name and also about whether I should make it passive (`updated` vs `update`), because of the immutability. But to be fair, none of the immutable Ruby methods are in passive form (e.g. `merge`).